### PR TITLE
Improve logging of config

### LINKED ISSUE #1354: Improve logging of config
@cket and I think it would be better to just log the config as one long line rather than multiple lines.

Also, http://jenkins.cgcloud.info/job/toil-pull-requests/2299/testReport/junit/src.toil.test.src.hotDeploymentTest/HotDeploymentTest/testDeferralWithFailureAndEncapsulation/ shows that config seems to logged twice. Search for `x/l/jobsv6IB5    ` in above console output.

### PR DESCRIPTION
**Pull request by natanlao**
Resolves #1354:
1. Config is now logged on one long line instead of on several lines
2. Fixed bug where config would be logged twice

### CODE REVIEW

**Comment by natanlao** [comment]
@hannes-ucsc I believe that this PR does fix the issue - I was able to reproduce the bug on 9feb01 and confirmed that 6155a logs config only once.

**Comment by natanlao** [comment]
Jenkins, test this please

**Comment by hannes-ucsc** [comment]
Could you explain why your change fixes the problem?

The IT logs corroborate your claim:

http://jenkins.cgcloud.info/job/toil-pull-request-integration-tests/648/testReport/src.toil.test.provisioners.cgcloud.provisionerTest/CGCloudRNASeqTest/testAutoScaledCluster/

Search for `82f6b9c6-81d2-4021-a729-3bff38dd3ce2    Next available file descriptor: 7`

There are still some double-logged lines which makes me think that the root of the problem lies deeper.

**Comment by natanlao** [comment]
@hannes-ucsc Ready for review

**Comment by natanlao** [comment]
Jenkins, test this please

**Comment by natanlao** [comment]
@hannes-ucsc @cket Ready for review

File: src/toil/__init__.py [hunk 1]
```diff
@@ -141,9 +140,9 @@ def lookupEnvVar(name, envName, defaultValue):
...
 
-def logProcessContext(config, logger):
+def logProcessContext(config):
     # toil.version.version (string) canont be imported at top level because it conflicts with
     # toil.version (module) and Sphinx doesn't like that.
     from toil.version import version
-    logger.info("Running Toil version %s.", version)
-    logger.info("Configuration:\n %s", pformat(config.__dict__))
+    log.info("Running Toil version %s.", version)
+    log.info("Configuration: %s", config.__dict__)
```
Comment (cket) [inline src/toil/__init__.py]: One thing, can you change this line to debug? You don't have to squash/rebase the commit to change this

**Comment by natanlao** [comment]
Jenkins, test this please

**Comment by cket** [comment]
@natanlao sorry, I wasn't clear enough. Only the config should be logged at debug level, the version should still be info.

**Pull request merged**